README.md
# N27 Smartphone Bank

Entwicklung einer Smartphone Bank nach dem Vorbild der N26 Bank

## Was soll die App können?

* Login
* Überweisen
* Kontostand abrufen
* Konto anlegen
* Hilfestellung / Support Daten / FAQ / Chat
* Kontobewegungen anzeigen
* alle Konten in übersicht anzeigen
* diverse Einstellungen
* Kredit aufnehmen
* Wertpapierkauf / - verkauf
* Geldanlage
* Kategorien bilden / Ausgaben auswerten
* Daueraufträge erstellen
* Kreditlinie festlegen

## Welche Dateien, die die Ansicht zeigen sollen, werden benötigt?

Alle Dateien, die etwas Anzeigen sollen werden in HTML geschrieben
* Startseite (index.html)
* Login (login.html)
  * Benutzer-ID-Eingabe-Feld
  * Passwort-Eingabe-Feld
  * Bestätigungs-Button
  


changelog.md
#Changelog

## 20190225

* heroku konfiguriert

## 20190218

* Favicon eingebaut
* login ausprobiert 
* npm-module installiert 
* 

Training/if-else-training.js
/**
 * Hier können Sie IF und ELSE trainieren.
 * 
 * Am besten dazu den nodemon starten:
 *      .\node_modules\.bin\nodemon Training\if-else-training.js
 * 
 * Dann die Ausgabe im Terminal beobachten.
 */


// Deklaration und Initialisierung 
// Lies "Der Wert 0 wird zugewiesen an eine Variable namens x"
 
let x = 0
 let y = 1

 if("GW11A".endsWith("A")||"GW11A".endsWith("B")||"GW11A".startsWith("GT")){
     console.log("Die Prüfung ist WAHR. Es werden die Anweisungen im Rumpf von IF ausgeführt")
 }else{
    console.log("Die Prüfung ist FALSCH. Es werden die Anweisungen im Rumpf von ELSE ausgeführt")
 }

 let email = ""

 if(email === ""){
     console.log("Der Schüler hat keine eMail-Adresse")
 }

// FÄLLE:

// 22. if("GW11A".endsWith("A")||"GW11A".endsWith("B")) Der Ausdruck ist wahr. 

// 21. if("GW11A".endsWith("A"))    Der Ausdruck ist wahr. Die Funktion prüft, ob eine Zeichenkette mit einer anderen Zeichenkette endet.

// 20. if("GW11A".includes("11"))   Der Ausdruck ist wahr. Die Funktion prüft, ob eine Zeichenkette eine andere Zeichenkette enthält.

// 19. if("GW11A".startsWith("GW")) Der Ausdruck ist wahr. Die Funktion prüft, ob eine Zeichenkette mit einer anderen Zeichenkette beginnt.

// 18. if("GW11A".length === 5) Der Ausdruck ist wahr. Die eigenschaft length gibt die Anzahl der Zeichen einer Zeichenkette zurück.

// 17. if(undefined)    Der Ausdruck ist falsch. Ein Objekt ist undefined, wenn es nicht existiert.

// 16. if(1 > 2 || 1 == 1)  Der Ausdruck ist wahr, weil eine der beiden Prüfungen wahr ist.

// 15. if(1 > 2 && 1 == 1)  Der Ausdruck ist falsch, weil eine der beiden Prüfungen falsch ist. Alle mit && verketteten Prüfungen müssen wahr sein, damit die Prüfung insgesamt Wahr wird.

// 14. if(false)    Der Ausdruck ist falsch.

// 13. if(true)     Der Ausdruck ist wahr.

// 12. if(0)        Die Zahl 0 ist falsch.

// 11. if(123)      Zahlen sind immer wahr, ausgenommen der Zahl 0

// 10. if("")       leere Zeichenketten in javascript sind falsch.

// 9. if("ah")      nicht leere Zeichenketten in javascript sind wahr.

// 8. if(1 != 0)    Vergleichsoperator: "ungleich"      Der Ausdruck ist wahr.

// 7. if(1 <= 1)    Vergleichsoperator: "kleiner gleich" Der Ausdruck ist wahr.

// 6. if(1 >= 1)    Vergleichsoperator: "größer gleich" Der Ausdruck ist wahr.

// 5. if(1 === "1") Vergleichsoperator: "gleich"      Der Ausdruck ist falsch.
//                  Das dreifache "===" des Wertes und des Typs.
//                  "1" : String / Zeichenkette
//                   1  : Zahl

// 4. if(1 == "1") Vergleichsoperator: "gleich"       Der Ausdruck ist wahr.
//                 Das doppelte "==" prüft auf Gleichheit des Wertes,
//                 aber nicht auf Gleichheit des Typs.

// 3. if(0 == 1)  Vergleichsoperator: "gleich"       Der Ausdruck ist falsch.

// 2. if(0 > 1)   Vergleichsoperator: "größer als"   Der Ausdruck ist falsch.

// 1. if(0 < 1)   Vergleichsoperator: "kleiner als"  Der Ausdruck ist wahr.

Training/klasse-objekt-ejs-trainieren.js
class Auto {
    constructor() {
        this.raeder
        this.sitze
    }
}

class Rechteck {
    constructor(){
        this.laenge
        this.breite
    }
}

class Footballer {
    constructor(){
        this.name 
        this.team
        this.position
        this.alter
        this.nummer
        this.groesse
        this.gewicht
        this.sacks
        this.tackles
        this.college
        this.fumbles
        this.touchdowns
    }
}
// Klassen definition für Schüler in einer Schule:





// Deklaration eines neuen objekts vom typ rechteck
// Deklaration = bekanntmachung
// let rechteck = ...

// Instanziierung eines neuen Objekts 
// Instanziierung erkennt man immer am reservierten Wort "new"
// Bei der Instanziierung wird Arbeitspeicher bereitgestellt
// ... = new Rechteck()

let rechteck = new Rechteck()
let schueler = new Schueler()
let footballer = new Footballer()
// 3. Intitialisierung (konkrete Eigenschaftswerte werden zugewiesen)

rechteck.breite = 2
rechteck.laenge = 3



footballer.name = "J.J. Watt"
footballer.team = "Houston Texans"
footballer.position = "RE"
footballer.alter = "30"
footballer.nummer = "99"
footballer.groesse = "1,96m"
footballer.gewicht = "131kg"
footballer.sacks = "92,0"
footballer.tackles = "455"
footballer.college = "Wisconsin"
footballer.fumbles = "22"
footballer.touchdowns = "5"

console.log("länge: " + rechteck.laenge)
console.log("breite: " + rechteck.breite)
const express = require('express')
const bodyParser = require('body-parser')
const app = express()
app.set('view engine', 'ejs')
app.use(bodyParser.urlencoded({extended: true}))
app.set('views', 'Training')

const server = app.listen(process.env.PORT || 3000, () => {
    console.log('Server lauscht auf Port %s', server.address().port)    
})

// Wenn localhost:3000/klasse-objekt-ejs-trainieren aufgerufen wird ...

app.get('/klasse-objekt-ejs-trainieren',(req, res, next) => {   

    // ... wird klasse-objekt-ejs-trainieren.ejs gerendert:

    res.render('klasse-objekt-ejs-trainieren', {   
        breite : rechteck.breite,
        laenge : rechteck.laenge,
        geschlecht : schueler.geschlecht,
        name : footballer.name,
        team : footballer.team,
        position : footballer.position,
        alter : footballer.alter,
        nummer : footballer.nummer,
        groesse : footballer.groesse,
        gewicht : footballer.gewicht,
        sacks : footballer.sacks,
        tackles : footballer.tackles,
        college : footballer.college,
        fumbles : footballer.fumbles,
        touchdowns : footballer.touchdowns



    })
})


server.js
class Konto {
    constructor() {
        this.Kontonummer
        this.Kontoart
    }
}

const express = require('express')
const bodyParser = require('body-parser')
const cookieParser = require('cookie-parser')
const app = express()
app.set('view engine', 'ejs')
app.use(express.static('public'))
app.use(bodyParser.urlencoded({extended: true}))
app.use(cookieParser())

const server = app.listen(process.env.PORT || 3000, () => {
    console.log('Server lauscht auf Port %s', server.address().port)    
})

app.get('/',(req, res, next) => {   

    let idKunde = req.cookies['istAngemeldetAls']
    
    if(idKunde){
        console.log("Kunde ist angemeldet als " + idKunde)
        res.render('index.ejs', {                              
        })
    }else{
        res.render('login.ejs', {                    
        })    
    }
})

app.get('/login',(req, res, next) => {         
    res.cookie('istAngemeldetAls', '')       
    res.render('login.ejs', {                    
    })
})

app.post('/',(req, res, next) => {   
    
    const idKunde = req.body.idKunde
    const kennwort = req.body.kennwort
        
    if(idKunde === "4711" && kennwort === "123"){            
        console.log("Der Cookie wird gesetzt:")
        res.cookie('istAngemeldetAls', idKunde)
        res.render('index.ejs', {           
        })
    }else{            
        console.log("Der Cookie wird gelöscht")
        res.cookie('istAngemeldetAls','')
        res.render('login.ejs', {                    
        })
    }
})
// Wenn die Seite localhost:3000/impressum aufgerufen wird, ...
app.get('/impressum',(req, res, next) => {   

    let idKunde = req.cookies['istAngemeldetAls']
    
    if(idKunde){
        console.log("Kunde ist angemeldet als " + idKunde)

        // ... Dann wird Impressum.ejs gerendert.

        res.render('impressum.ejs', {                              
        })
    }else{
        res.render('login.ejs', {                    
        })    
    }
})

// Wenn die Seite localhost:3000/kontoAnlegen angesurft wird, dann wir kontoAnlegen.ejs gerendert


app.get('/kontoAnlegen',(req, res, next) => {   

    let idKunde = req.cookies['istAngemeldetAls']
    
    if(idKunde){
        console.log("Kunde ist angemeldet als " + idKunde)

        // ... Dann wird Impressum.ejs gerendert.

        res.render('kontoAnlegen.ejs', {   
            meldung : ""                           
        })
    }else{
        res.render('login.ejs', {                    
        })    
    }
})

// Wenn der Button auf der kontoanlegen Seite gedrückt wird, dann wird kontoAnlegen.ejs gerendert.

app.post('/kontoAnlegen',(req, res, next) => {   

    let idKunde = req.cookies['istAngemeldetAls']
    
    if(idKunde){

        let konto = new Konto()
        konto.Kontonummer = req.body.kontonummer
        konto.Kontoart = req.body.kontoart

        res.render('kontoAnlegen.ejs', {  
            meldung : "Das "+ konto.Kontoart + " " + konto.Kontonummer + " wurde erfolgreich angelegt."                            
        })
    }else{
        res.render('login.ejs', {                    
        })    
    }
})
